experiments/rtp-hdrext/transport-wide-cc-02/index.md

```diff
@@ -0,0 +1,68 @@
+---
+layout: default
+title: transport-wide-cc-02
+permalink: /experiments/rtp-hdrext/transport-wide-cc-02/
+---
+
+This RTP header extension is an extended version of the extension defined in
+<https://tools.ietf.org/html/draft-holmer-rmcat-transport-wide-cc-extensions-01> 
+
+**Name:** "Transport-wide congenstion control 02"
+
+**Formal name:**
+<http://www.webrtc.org/experiments/rtp-hdrext/transport-wide-cc-02>
+
+**Status:** This extension is defined here to allow for experimentation. Once
+experience has shown that it is useful, we intend to make a proposal based on
+it for standardization in the IETF.
+
+The original extension defines a transport-wide sequence number that is used in
+feedback packets for congestion control. The original implementation sends these
+feedback packets at a periodic interval. The extended version presented here has
+two changes compared to the original version:
+* Feedback is sent only on request by the sender, therefore, the extension has
+  two optional bytes that signals that a feedback packet is requested.
+* The sender determines if timing information should be included or not in the
+  feedback packet. The original version always include timing information.
+
+Contact <kron@google.com> or <sprang@google.com> for more info.
+
+## RTP header extension format
+
+### Data layout overview
+Data layout of transport-wide sequence number
+     1-byte header + 2 bytes of data:
+
+      0                   1                   2 
+      0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3
+     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     |  ID   | L=1   |transport-wide sequence number |
+     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+Data layout of transport-wide sequence number and optional feedback request
+     1-byte header + 4 bytes of data:
+
+      0                   1                   2                   3
+      0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     |  ID   | L=3   |transport-wide sequence number |T|  seq count  |
+     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     |seq count cont.|
+     +-+-+-+-+-+-+-+-+
+
+### Data layout details
+The data is written in the following order,
+* transport-wide sequence number (16-bit unsigned integer)
+* feedback request (optional) (16-bit unsigned integer)<br>
+  If the extension contains two extra bytes for feedback request, this means
+  that a feedback packet should be generated and sent immediately. The
+  feedback request consists of a one-bit flag T and a 15-bit unsigned integer
+  seq count that are merged as: (T << 15) + seq count.
```

Comment:
This can more easily be expressed as "an one-bit field giving the flag value T and a 15-bit field giving the sequence count as an unsigned number".

Are you really going to support requesting 32767 packets of feedback? What happens if the receiver has already dropped history for packet 12 and beyond? Is there a minimum required history size?

Reply:
Updated according to you suggestion.

The protocol will at least support that. If running at really high bit rates (say 100 Mbps), that would correspond to roughly 3 seconds of packets. Not sure if 32767 packets will ever be requested, but I don't think it's unreasonable to request feedback in the order of 1000 packets. The current implementation in WebRTC should be able to handle this. There's a maximum limit of 32767 packets in the buffer and the buffer will be cleared based on the received feedback requests. For example, if the sender requests feedback from packet 1000 to packet 1200, this is an indication to the receiver that it can clear everything up to packet 999.